frontend/src/components/class/ClassList.tsx

```diff
@@ -129,36 +141,81 @@ const ClassList = () => {
-        <DropdownMenu
-          trigger={
-            <Button
-              variant={ButtonVariant.secondary}
-              onClick={() => {
-                router.push(`/class/${rowData.id}/edit`);
-              }}
-              data-testid={`class-${rowData.id}-edit-button`}
-            >
-              <FontAwesomeIcon icon={faEdit} />
-            </Button>
-          }
-        >
-          <DropdownMenu.Item
-            onClick={() => {
-              setClassIdToDelete(rowData.id);
-              setShowDeleteConfirmationModal(true);
-            }}
-            data-testid={`class-${rowData.id}-delete-button`}
-          >
-            {intl.formatMessage(TableMessages.delete)}
-          </DropdownMenu.Item>
-        </DropdownMenu>
-      </div>
-    ),
-    [router, intl],
-  );
+  const dataWithMockFields = data?.map((classItem) => ({
+    ...classItem,
+    degree: classItem.degree || "12th grade",
+    schoolYear: classItem.schoolYear || "2024-2025",
+    status: classItem.status || "current",
+  }));
+
+  const columns: ColumnDef<ExistingClassWithTeacher>[] = [
+    {
+      accessorKey: "name",
+      header: intl.formatMessage(messages.nameColumn),
+      cell: (info) => (
+        <span data-testid={`class-${info.row.original.id}-name`}>
+          {info.row.original.name}
+        </span>
+      ),
+      meta: {
+        columnType: ColumnType.text,
+      },
+    },
+    {
+      accessorKey: "degree",
+      header: intl.formatMessage(messages.degreeColumn),
+      enableSorting: false,
+      cell: (info) => {
+        const degree = info.row.original.degree;
+        return <span>{degree}</span>;
+      },
+      meta: {
+        columnType: ColumnType.text,
+      },
+    },
+    {
+      accessorKey: "schoolYear",
+      header: intl.formatMessage(messages.schoolYearColumn),
+      enableSorting: false,
+      cell: (info) => {
+        const schoolYear = info.row.original.schoolYear;
+        return <span>{schoolYear}</span>;
+      },
+      meta: {
+        columnType: ColumnType.text,
+      },
+    },
+    {
+      accessorKey: "teacher",
+      header: intl.formatMessage(messages.teacherColumn),
+      enableSorting: false,
+      cell: (info) => {
+        const teacher = info.row.original.teacher;
+        return <span>{teacher?.name || "Mr. Bumbacher"}</span>;
+      },
+      meta: {
+        columnType: ColumnType.text,
+      },
+    },
+    {
+      accessorKey: "status",
+      header: intl.formatMessage(messages.statusColumn),
+      enableSorting: false,
+      cell: (info) => {
+        const status = info.row.original.status;
+        const isActive = status === "current";
+        return (
+          <StatusWrapper>
+            <FaCircle color={isActive ? "#22c55e" : "#6b7280"} size={10} />
+            <span>{intl.formatMessage(getClassStatusMessage(status))}</span>
+          </StatusWrapper>
+        );
+      },
+      meta: {
+        columnType: ColumnType.text,
+      },
+    },
```

Comment:
Per our discussion on Nov 6, we don't need these extra columns (degree, schoolYear, status)